internal/webhook/v1alpha1/biosversion_webhook_test.go

```diff
@@ -0,0 +1,224 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	metalv1alpha1 "github.com/ironcore-dev/metal-operator/api/v1alpha1"
+	// TODO (user): Add any additional imports if needed
+)
+
+var _ = Describe("BIOSVersion Webhook", func() {
+	var (
+		obj       *metalv1alpha1.BIOSVersion
+		oldObj    *metalv1alpha1.BIOSVersion
+		validator BIOSVersionCustomValidator
+	)
+
+	BeforeEach(func() {
+		obj = &metalv1alpha1.BIOSVersion{}
+		oldObj = &metalv1alpha1.BIOSVersion{}
+		validator = BIOSVersionCustomValidator{Client: k8sClient}
+		Expect(validator).NotTo(BeNil(), "Expected validator to be initialized")
+		Expect(oldObj).NotTo(BeNil(), "Expected oldObj to be initialized")
+		Expect(obj).NotTo(BeNil(), "Expected obj to be initialized")
+		// TODO (user): Add any setup logic common to all tests
+	})
+
+	AfterEach(func() {
+		// TODO (user): Add any teardown logic common to all tests
+	})
+
+	Context("When creating or updating BIOSVersion under Validating Webhook", func() {
+		It("Should deny creation if a Spec.ServerRef field is duplicate", func(ctx SpecContext) {
+			By("Creating an BIOSVersion")
+			biosVersionV1 := &metalv1alpha1.BIOSVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    "ns.Name",
+					GenerateName: "test-",
+				},
+				Spec: metalv1alpha1.BIOSVersionSpec{
+					Version:                 "P71 v1.45 (12/06/2017)",
+					Image:                   metalv1alpha1.ImageSpec{URI: "asd"},
+					ServerRef:               &v1.LocalObjectReference{Name: "foo"},
+					ServerMaintenancePolicy: metalv1alpha1.ServerMaintenancePolicyEnforced,
+				},
+			}
```

Comment:
Would it make sense to setup `biosVersionV1` in `BeforeEach()`? It seems to be the same in all test cases.